stubs/Relation.stub

```diff
@@ -2,19 +2,132 @@
 
 namespace Illuminate\Database\Eloquent\Relations;
 
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+
 /**
- * @template TRelatedModel of \Illuminate\Database\Eloquent\Model
+ * @template TRelatedModel of Model
+ * @template TOwningModel of Model
+ * @template TResult
```

Comment:
Hello @nagmat84!
What would Ms. Laskov says to current Larastan users writing `Relation<AModel>` only?

Reply:
I haven't tried yet but I expect it to be backward compatible. That is also the reason why the order of template parameters is `TRelatedModel`, `TOwningModel` and `TResult` with the previously existing parameter being the first although I personally consider the order `TOwningModel`, `TRelatedModel` and `TResult` to be more intuitive.

An unbound `TOwningModel` will be resolved to `Eloquent\Model`, because this is the upper bound.

For all users which do not implement custom `Relation` classes nothing will change. Moreover, the methods `hasMany`, `hasOne`, etc., are handled such that the template parameter is automatically bound to the type of the calling class.

For users which implement custom `Relation`-classes there might be changes. However I expect them to fall into two categories. Users who do not overwrite the methods which make use of `TOwningClass` won't be affected. For them it will simply be `Eloquent\Model`. Users who overwrite the methods which make use of `TOwningClass` will notice a difference. However this category of users should be suffering from the same bug as me. So they should be happy. Moreover, I assume this group to be small. Otherwise someone should already have noticed the bug before me.

Reply:
> I haven't tried yet but I expect it to be backward compatible.

In a later stage please test this.